src/app/services/cities/cities.service.ts

```diff
@@ -1,20 +1,33 @@
 import { HttpClient } from '@angular/common/http';
 import { Injectable } from '@angular/core';
-import { Observable } from 'rxjs';
-import { map } from 'rxjs/operators';
+import { Observable, of } from 'rxjs';
+import { catchError, map, shareReplay } from 'rxjs/operators';
 import { API } from 'src/app/constants';
+import { MatSnackBar } from '@angular/material/snack-bar';
 
 @Injectable({
   providedIn: 'root',
 })
 export class CitiesService {
-  constructor(private http: HttpClient) {}
+  getAllObservableCache: Observable<any> | null = null;
+  constructor(private http: HttpClient, private snackBar: MatSnackBar) {}
 
   getAll(): Observable<any> {
-    return this.http.get(`${API}/cities?levels=3`).pipe(
-      map((data) => {
-        return data;
-      })
-    );
+    if (!this.getAllObservableCache)
+      this.getAllObservableCache = this.http.get(`${API}/cities?levels=3`).pipe(
+        map((data) => {
+          return data;
+        }),
+        shareReplay(1),
+        catchError(() => {
+          this.snackBar.open(
+            'Houve um erro buscando as cidades disponíveis! Tente novamente mais tarde.',
+            'Fechar'
+          );
+          return of({ cities: [] });
+        })
```

Comment:
brabooo 🔥 

pra fins de debug interno eu só adicionaria um `console.error`.
```suggestion
        catchError((error) => {
          console.error(error);
          this.snackBar.open(
            'Houve um erro buscando as cidades disponíveis! Tente novamente mais tarde.',
            'Fechar'
          );
          return of({ cities: [] });
        })
```

Reply:
Feito!